src/ethereum_test_specs/eof.py

```diff
@@ -228,7 +229,17 @@ def make_eof_test_fixture(
             if vector.container_kind == ContainerKind.INITCODE:
                 args.append("--initcode")
             result = eof_parse.run(*args, input=str(vector.code))
-            self.verify_result(result, expected_result, vector.code)
+            if self.no_expectations_on_validity:
```

Comment:
I have an alternative/extension to this flag. I think it would be good that either:
1/ `no_expectations_on_validity` flag is combined with `expect_exception`, so that either the code is valid OR the exception is one of the expected ones. Then ALL of the `expect_exception` ones would make it into the fixture, so that clients are not restricted to a particular expected error. In the case of the RJUMP*/CALLF tests those expected exceptions would be the stack related ones
2/ remove the new flag and allow `None` to be an element of the `expect_exception` list thereby admitting that the code may validate successfully OR throw one of the exceptions in the list. This has the disadvantage of requiring the support of this by the clients test runners

The motivation behind would be to safe guard against the test suddenly generating code which is EOF-invalid for an unrelated cause (say, undefined instruction or whatever), making the whole thing less likely to break silently.

I wonder what be your preference (or neither option is OK or even the whole concept is not acceptable)?

I have an implementation of (1/), but I haven't yet fully investigated its consequences.

Reply:
After sleeping on this I realized (1/) is both the better approach and necessary, [`Close the catalogue of potential exceptions`](https://github.com/ethereum/execution-spec-tests/pull/833/commits/e90423631387d382e370858d70da12ae9340b207) handles that, PTAL

Reply:
I still think that (1/) is too loose in the sense that we could miss detecting an important change and the tests passing inadvertently, that's my main concern.

I think we should analyze how hard it would be to isolate the valid containers from the invalid ones in the parametrization.

Reply:
Thanks for taking a look! Yea, I did some thinking about this and it seemed unfeasible / too complex. But, actually you gave me an idea here - maybe I've been trying too hard to make the produced code valid. What I could instead focus on is generate the code as it is and add some heuristics which could narrow down the catalogue even further. Like exception A can only happen if condition X is met, exception B if condition Y, etc. That would also isolate valid from invalid, like you suggest.

Maybe those 3 remaining exceptions can be "predicted" easily enough, lemme try that

Reply:
@marioevz I pushed the version which "predicts" validity based on some traits of the code. I didn't yet think about refactoring those traits into some kind of class maybe, to avoid the massive ifs (also not sure about my python-fu here), but here it is, LMK if this looks better now.

Reply:
@marioevz Did you have a chance to look at the changes? :point_up: